caps/capabilities.go

```diff
@@ -33,12 +33,63 @@ type Capability struct {
 	// the behavior allowed and expected from providers and consumers of that
 	// capability, and also which information should be exchanged by these
 	// parties.
-	Type Type
+	Type *Type
 	// Attrs are key-value pairs that provide type-specific capability details.
 	Attrs map[string]string
 }
 
+var (
+	testCapability = &Capability{
+		Name:  "test-name",
+		Label: "test-label",
+		Type:  testType,
+		Attrs: nil,
+	}
+)
+
 // String representation of a capability.
 func (c Capability) String() string {
 	return c.Name
 }
+
+// CapabilityRepr is the JSON representation of a capability.
+// It exists so that Type can be replaced by Type.Name as we don't want to
+// create or describe capabilities fully, just to refer to them.
```

Comment:
It's not clear to me what this is saying. Gut feeling is that this is very strange language to see in a public API. Can't say so before understanding, but most likely something to be tuned here.